compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -209,6 +211,25 @@ private void writeFunction(ByteBuf buf, BIRTypeWriter typeWriter, BIRNode.BIRFun
         // Function type as a CP Index
         BIRWriterUtils.writeType(cp, buf, birFunction.type);
 
+
+        // Children functions
+        if (birFunction.type.tsymbol == null || birFunction.type.tsymbol.owner == null) {
```

Comment:
Shalll we wrap this `birFunction.type.tsymbol` into a new variable.

Reply:
Modified in commit #a0abba10a989aee9ad4c0ab14369cbeff55c8907